de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/M_Element.java

```diff
@@ -321,13 +321,67 @@ protected boolean afterSave(boolean newRecord, boolean success)
 				log.debug("PrintFormatItem updated #" + no);
 			}
 
+			// AD_Tab
+			if (is_ValueChanged(I_AD_Element.COLUMNNAME_Name)
+					|| is_ValueChanged(I_AD_Element.COLUMNNAME_Description)
+					|| is_ValueChanged(I_AD_Element.COLUMNNAME_Help)
+					|| is_ValueChanged(I_AD_Element.COLUMNNAME_CommitWarning))
+			{
+
+				sql = new StringBuffer("UPDATE AD_Tab SET Name=")
+						.append(DB.TO_STRING(getName()))
+						.append(", Description=").append(DB.TO_STRING(getDescription()))
+						.append(", Help=").append(DB.TO_STRING(getHelp()))
+						.append(", ").append(I_AD_Element.COLUMNNAME_CommitWarning).append(" = ").append(DB.TO_STRING(getCommitWarning()))
+						.append(" WHERE AD_Element_ID = ").append(get_ID());
+
+				no = DB.executeUpdate(sql.toString(), get_TrxName());
+				log.debug("Tabs updated #" + no);
+			}
```

Comment:
why not using IQueryBuilder instead of hardcoded SQL?

Reply:
same for those SQLs below...

Reply:
I will leave them as they are, as we discussed.
The reason is that our org.compiere.model.IQuery.updateDirectly() is not good for queries that will get into a migration script.